packages/base-map/src/index.js

```diff
@@ -0,0 +1,271 @@
+  componentDidMount() {
+    const lmap = this.refs.map.leafletElement;
+    lmap.options.singleClickTimeout = 250;
+    lmap.on("singleclick", this.onLeftClick);
+  }
+
+  componentDidUpdate() {}
+
+  // remove custom overlays on unmount
+  // TODO: Is this needed? It may have something to do with mobile vs desktop views
+  componentWillUnmount() {
+    const lmap = this.refs.map.leafletElement;
+    lmap.eachLayer(layer => {
+      // Do not inline, there is a 'this' implied.
+      lmap.removeLayer(layer);
+    });
+  }
+
+  onLeftClick = e => {
+    const { onClick } = this.props;
+    if (typeof onClick === "function") onClick(e);
+  };
+
+  render() {
+    const {
+      baseLayers,
+      center,
+      children,
+      maxZoom,
+      popupLocation,
+      popupContent,
+      onPopupClosed,
+      onOverlayAdded,
+      onOverlayRemoved,
+      onViewportChanged,
+      zoom
+    } = this.props;
+
+    // Separate overlay layers into user-controlled (those with a checkbox in
+    // the layer control) and those that are needed by the app (e.g., stop viewer
+    // and itinerary overlay).
+    const userControlledOverlays = [];
+    const fixedOverlays = [];
+    React.Children.toArray(children).forEach(child => {
+      if (child.props.name) userControlledOverlays.push(child);
+      else fixedOverlays.push(child);
+    });
+
+    return (
+      <Map
+        ref="map"
+        className="map"
+        center={center}
+        zoom={zoom}
+        maxZoom={maxZoom}
+        // onClick={this.onLeftClick}
+        // Note: Map-click is handled via single-click plugin, set up in componentDidMount()
+        onOverlayAdd={onOverlayAdded}
+        onOverlayRemove={onOverlayRemoved}
+        onViewportChanged={onViewportChanged}
+      >
+        {/* Create the layers control, including base map layers and any
+         * user-controlled overlays. */}
+        <LayersControl position="topright">
+          {/* base layers */}
+          {baseLayers &&
+            baseLayers.map((layer, i) => {
+              return (
+                <LayersControl.BaseLayer
+                  name={layer.name}
+                  checked={i === 0}
+                  key={i}
+                >
+                  {
+                    // Fix tile size/zoom offset: https://stackoverflow.com/a/37043490/915811
+                    // Also, split the declaration between the two
+                    // outcomes to avoid error 'attempted to load an infinite number of tiles'
+                    // if the tiles cannot be loaded for any reason.
+                  }
+                  {L.Browser.retina && layer.hasRetinaSupport ? (
+                    <TileLayer
+                      url={layer.url}
+                      attribution={layer.attribution}
+                      maxZoom={layer.maxZoom}
+                      tileSize={512}
+                      zoomOffset={-1}
+                      detectRetina
+                    />
+                  ) : (
```

Comment:
I prefer for multi-line ternaries to be styled like:
```
{L.Browser.retina && layer.hasRetinaSupport
  ? x
  : y
}
```
If what you have is what airbnb specifies, it's fine for now.

Reply:
The original format is enforced by es-lint prior to commit.